ios/engine/KMEI/KeymanEngine/Classes/Extension/UserDefaults+Types.swift

```diff
@@ -133,13 +133,35 @@ public extension UserDefaults {
     }
   }
   
-  func userLexicalModels(forLanguage lgCode: String) -> [InstallableLexicalModel]? {
-    if let models = userLexicalModels {
-      return models.compactMap { ($0.languageID == lgCode) ? $0 : nil }
+  // stores a dictionary of lexical model ids keyed to language ids, i.e., [langID: modelID]
+  var languageModelSelections: [String: String]? {
+    get {
+      return dictionary(forKey: Key.userPreferredLexicalModels) as? [String : String]
+    }
+    
+    set(lexicalModelPrefs) {
+      set(lexicalModelPrefs, forKey: Key.userPreferredLexicalModels)
+    }
+  }
+  
+  // returns lexical model id, given language id
+  func preferredLexicalModelID(forLanguage lgCode: String) -> String? {
+    if let modelPrefs = languageModelSelections {
+      return modelPrefs[lgCode]
```

Comment:
What will this do if `lgCode` is not found in `modelPrefs`?

Reply:
It will return nil, which is okay, as it's Optional and callers already handle that.